src/compile/layout.ts

```diff
@@ -86,7 +86,7 @@ function parseUnitSizeLayout(model: UnitModel, channel: Channel): SizeComponent
 }
 
 function unitSizeExpr(model: UnitModel, channel: Channel, nonOrdinalSize: number): string {
-  if (model.has(channel)) {
+  if (model.has(channel) || (channel === X && model.has(X2)) || (channel === Y && model.has(Y2))) {
```

Comment:
Yeah that makes sense. I tried it, its working. I will push that in the next commit